### PR TITLE
Add `.hasCSS()` assertion

### LINKED ISSUE #90: Add `assert.dom(...).hasCSS({ ... })` assertion
### Example

```js
assert.dom('.some-element').hasCSS({
  opacity: 1,
  width: 340,
});
```

### PR DESCRIPTION
**Pull request by selvagsz**
Add assertions for styles like 

```js
assert.dom('.some-element').hasCSS({
  opacity: 1,
  width: '340px',
});
```

Fixes #90

### CODE REVIEW

File: lib/__tests__/has-css.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  test('works for subset of css assertions', () => {
+    assert.dom('.foo').hasCSS({
+      opacity: '1',
+      width: '200px',
+      'text-align': 'center',
+    });
+    expect(assert.results).toEqual([{
+      actual: {opacity: '1', width: '200px', 'text-align': 'center'},
+      expected: {opacity: '1', width: '200px', 'text-align': 'center'},
```
[the remark this answers is not in the document]
Comment (scalvert) [reply lib/__tests__/has-css.ts]: `JSON.stringify` may be the answer here, I believe it may not be deterministic for non-integer keys.

**Comment by alexlafroscia** [comment]
I'd be really interested in helping get this merged -- is there anything I can do to help?

**Comment by Turbo87** [comment]
> is there anything I can do to help?

invent cloning or a time machine 😜 

thanks for the reminder, I'll try to have a look in the next days!

**Comment by Turbo87** [comment]
I just had a quick look, it seems that my comment above (https://github.com/simplabs/qunit-dom/pull/106#discussion_r209559133) is not addressed yet 🤔 

@alexlafroscia if you want to help with it you could investigate if it needs to be fixed and if so what the best way forward would be

**Comment by selvagsz** [comment]
Sorry for letting this stale.  I’ll take a look today

**Comment by selvagsz** [comment]
@Turbo87 this is how it renders on failure

image

File: dist/qunit-dom.js [hunk 2]
```diff
@@ -0,0 +1,1098 @@
+(function (QUnit) {
```
Comment (Turbo87) [inline dist/qunit-dom.js]: this should not be committed to the repo

File: lib/__tests__/has-style.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      width: '200px',
+      'text-align': 'center',
+    });
+    expect(assert.results).toEqual([{
+      actual: {opacity: '1', width: '200px', 'text-align': 'center'},
+      expected: {opacity: '1', width: '200px', 'text-align': 'center'},
+      message: 'Element .foo has style \"{\"opacity\":\"1\",\"width\":\"200px\",\"text-align\":\"center\"}\"',
+      result: true,
+    }]);
+  });
```
Comment (Turbo87) [inline lib/__tests__/has-style.ts]: can you add a few more tests similar to the tests we have for other assertions?

File: API.md [hunk 4]
```diff
@@ -531,10 +534,31 @@ Assert that the `value` property of an [HTMLInputElement][] is empty.
...
 ```javascript
 assert.dom('input.username').hasNoValue();
 ```
+
+### hasStyle
+
+Assert that the [HTMLElement][] has the `expected` style declarations using
+[`window.getComputedStyle`](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle).
+
+**Aliases:** `lacksValue`
```
Comment (Turbo87) [inline API.md]: where is this coming from?

File: API.md [hunk 5]
```diff
@@ -531,10 +534,31 @@ Assert that the `value` property of an [HTMLInputElement][] is empty.
 
 #### Parameters
 
--   `message` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)?** 
+-   `message` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)?**
```
Comment (Turbo87) [inline API.md]: where are these changes coming from? did you generate the file or manually adjust it in the editor?

File: lib/assertions/is-disabled.ts [hunk 6]
```diff
@@ -12,13 +13,21 @@ export default function isDisabled(message, options: { inverted?: boolean } = {}
...
+  )) {
+    let ariaDisabled = element.attributes['aria-disabled'];
+    if(ariaDisabled){
+      state = ariaDisabled.value === 'true' ? true : false;
+    } else {
+      throw new TypeError(`Generic Element Type: ${element.toString()} does not use aria-disabled attribute`);
+    }
+
+  } else {
+    state = element.disabled
```
Comment (Turbo87) [inline lib/assertions/is-disabled.ts]: this is an unrelated change that seems to somehow have ended up in here 🤔

**Comment by selvagsz** [comment]
Sorry for messing up this PR. Superseded by #168  
Closing this one
